pytato/array.py

```
@@ -1804,4 +1800,189 @@ def ones(namespace: Namespace, shape: ConvertibleToShape, dtype: Any = float,
+    """
+    Returns (x1 >= x2) element-wise.
+    """
+    return _compare(x1, x2, ">=")
+
+# }}}
+
+
+# {{{ logical operations
+
+def logical_or(x1: Union[Array, Number],
+               x2: Union[Array, Number]) -> Union[Array, bool]:
+    """
+    Returns the element-wise logical OR of *x1* and *x2*.
+    """
+    # https://github.com/python/mypy/issues/3186
+    import pytato.utils as utils
+    return utils.broadcasted_binary_op(x1, x2,
+                                       lambda x, y: prim.LogicalOr((x, y)),
+                                       lambda x, y: np.bool8)  # type: ignore
+
+
+def logical_and(x1: Union[Array, Number],
+               x2: Union[Array, Number]) -> Union[Array, bool]:
+    """
+    Returns the element-wise logical AND of *x1* and *x2*.
+    """
+    # https://github.com/python/mypy/issues/3186
+    import pytato.utils as utils
+    return utils.broadcasted_binary_op(x1, x2,
+                                       lambda x, y: prim.LogicalAnd((x, y)),
+                                       lambda x, y: np.bool8)  # type: ignore
+
+
+def logical_not(x: Union[Array, Number]) -> Union[Array, bool]:
+    """
+    Returns the element-wise logical NOT of *x*.
+    """
+    if isinstance(x, Number):
+        # https://github.com/python/mypy/issues/3186
+        return np.logical_not(x)  # type: ignore
+
+    from pytato.utils import with_indices_for_broadcasted_shape
+    return IndexLambda(x.namespace,
+                       with_indices_for_broadcasted_shape(prim.Variable("_in0"),
+                                                          x.shape,
+                                                          x.shape),
+                       shape=x.shape,
+                       dtype=np.bool8,
+                       bindings={"_in0": x})
+
+# }}}
+
+
+# {{{ where
+
+def where(condition: Union[Array, Number],
+          x: Union[Array, Number], y: Union[Array, Number]) -> Union[Array, Number]:
```

`where` actually also has a single-argument form. Maybe raise an intelligible error if that's what the user wanted?

---

Yep, had not accounted for that. Thanks! bf89f130908aa4e07bc50e4e721fa0ccebb31a6f